### DISCUSSION TITLE
[make-fetch-happen] omits cause argument for `onRetry` function

### DISCUSSION THREAD

**Discussion by jmpage** [DefinitelyTyped/DefinitelyTyped#62221] [Issues with a @types package]
The types for `make-fetch-happen` specify that it may be called with `onRetry` as an option, which aligns with its [README](https://github.com/npm/make-fetch-happen/blob/main/README.md#opts-onretry):

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5c248b832935aebece0abff07360102f1e9111f3/types/make-fetch-happen/index.d.ts#L104

As shown above, the type definition only allows a function with arity 0 to be provided. However, `make-fetch-happen` appears to always pass an argument to this function of one of the following types:

1. `Response` from `minipass-fetch` (which reimplements `node-fetch`). [code][1]
2. All errors raised as part of the fetch process ([code][2]), including:
    a. `AbortError` and `FetchError` from, and thrown by, `minipass-fetch`
    b. `Error` containing a `retried` property, thrown by `promise-retry`

Including and typing this argument would be helpful for developers who want to correctly utilize this functionality provided by `make-fetch-happen`.

Would it be reasonable to open a PR to add this argument to the `onRetry` type definition?

[1]: https://github.com/npm/make-fetch-happen/blob/ea711108e53f04d848084db51477b5a24d176884/lib/remote.js#L83
[2]: https://github.com/npm/make-fetch-happen/blob/ea711108e53f04d848084db51477b5a24d176884/lib/remote.js#L106

[a comment by an automated account is not shown]

**Comment by trevor-scheer** [comment]
Thanks @jmpage, that would be a welcome PR!

**Comment by Semigradsky** [comment]
What about?
```ts
onRetry?: ((cause: unknown) => void) | undefined;
```

**Reply by trevor-scheer** [reply]
Agree with this, caught errors should be treated as `unknown`.

**Comment by MysteryBlokHed** [comment]
I just opened #63379
